lighthouse-cli/test/smokehouse/lighthouse-runners/bundle.js

```diff
@@ -8,47 +8,75 @@
  * @fileoverview  A runner that launches Chrome and executes Lighthouse via a
  * bundle to test that bundling has produced correct and runnable code.
  * Currently uses `lighthouse-dt-bundle.js`.
+ * Runs in a worker to avoid messing up marky's global state.
  */
 
 import fs from 'fs';
+import os from 'os';
+import {Worker, isMainThread, parentPort, workerData} from 'worker_threads';
 
 import puppeteer from 'puppeteer-core';
 import ChromeLauncher from 'chrome-launcher';
 
 import ChromeProtocol from '../../../../lighthouse-core/gather/connections/cri.js';
 import {LH_ROOT} from '../../../../root.js';
+import {loadArtifacts, saveArtifacts} from '../../../../lighthouse-core/lib/asset-saver.js';
 
-const originalBuffer = global.Buffer;
-const originalRequire = global.require;
-if (typeof globalThis === 'undefined') {
-  // @ts-expect-error - exposing for loading of dt-bundle.
-  global.globalThis = global;
+// This runs only in the worker. The rest runs on the main thread.
+if (!isMainThread && parentPort) {
+  (async () => {
```

Comment:
technically don't we have top-level await? We might not want to break that seal yet, though

Reply:
https://github.com/GoogleChrome/lighthouse/issues/13964